### ISSUE TITLE
[Proposal] Allow to map table column names to Eloquent attributes

### ISSUE THREAD

**Issue by mnabialek** [laravel/ideas#1802]
I often work with databases that have names of columns I cannot change. Often they are in strange format (for example instead of `id` there is `usr_id`, instead of `email` there is `usr_email`) or they have typos.

It would be great if it was possible to set mappings for Eloquent. So for example if I use:

`User::find(5)->email`

I would get value of `usr_email` column. Same for assignment and using `toArray()` or relationships.

At the moment I haven't found really good way to achieve it without any drawbacks.

**Comment by staudenmeir** [comment]
Have you seen https://github.com/jarektkaczyk/eloquence-mappable?

**Comment by mnabialek** [comment]
No, I haven't seen it, maybe I will try it in next project, however it would be great is it was built into core
